lms/djangoapps/mobile_api/user_sharing_pref/views.py

```diff
@@ -0,0 +1,56 @@
+"""
+Views for users sharing preferences
+"""
+
+from rest_framework import generics, permissions, status 
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.response import Response
+from openedx.core.djangoapps.user_api.api.profile import preference_info, update_preferences
+import serializers
+
+
+class UserSharing(generics.ListCreateAPIView):
+    """
+    **Use Case**
+
+        An API to retrieve or update the users social sharing settings
+
+    **GET Example request**:
+
+        GET /api/mobile/v0.5/settings/share_pref/ 
+
+    **GET Response Values**
+
+        {'share_pref': 'True'}
```

Comment:
This should be specific to `share_with_facebook_friends`.

In the future, if we implement other social integrations, the user may be selective about which social groups to share with.  For example, share with Facebook friends, but not with Google+ friends.